### ISSUE TITLE
[Bug] Unity 2021.2 "Internal build system error. Backend exited with code" error after import Firebase SDK 8.8.0

### ISSUE THREAD

**Issue by sableangle** [firebase/quickstart-unity#1228]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: Unity 2021.2.9f1
  * Firebase Unity SDK version: 8.8.0
  * Source you installed the SDK:  Unity Package Manager by point to the path where tar file is.
  (tar file and how to install SDK with tar file is follow the guide here: https://developers.google.com/unity/instructions#install-tgz)
  * Problematic Firebase Component: com.google.firebase.app (Auth, Database, etc.)
  * Other Firebase Components in use: Auth, Database, Storage, messaging, remote-config, crashlytics, analytics (Auth, Database, etc.)
  * Additional SDKs you are using: External dependency manager (Facebook, AdMob, etc.)
  * Platform you are using the Unity editor on: MacBook Pro with M1 chip and OS version is 12.3 Beta （The OS version may not a reason due to the same issue occur on macOS 12.2）
  * Platform you are targeting: Android/iOS (iOS, Android, and/or desktop)
  * Scripting Runtime: IL2CPP (Mono, and/or IL2CPP)

### [REQUIRED] Please describe the issue here:

(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)

Once I import the Firebase SDK the Unity Editor show the message likes "Internal build system error. Backend exited with code", "Backend produced unparsable log entry" and then the compile backend of Unity Editor is totally crash so I cannot compile my game script after the script is modified.

#### Steps to reproduce:

Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?
What's the issue repro rate? (eg 100%, 1/5 etc)

1. Create empty Unity project and import Firebase SDK by tar file and EDM4U. (Basically I just import "com.google.firebase.app" package and issue comes)
2. Create a new .cs script and do some modify on the script.
3. Issue reproduce.
4. 100% reproduce.
5. Reproduced on Unity 2021.2.9, 2021.2.10

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

#### Relevant Code:

```
// TODO(you): code here to reproduce the problem
```

**Comment by sableangle** [comment]
Extra information:
I also got error looks like #835 

I'm sure the python is installed in the computer

When I tested the python install status by those command I got the result like this.
` 
which python
/opt/homebrew/bin/python

python --version
Python 3.9.10
`

And some updates, after deleting the file /Asstes/Firebase/Editor/Firebase.Editor.dll the error message is gone and everything on Editor works fine. (Not tested the Firebase is works great or not on the built player)

**Comment by AlmostMatt** [comment]
Unfortunately Mac with M1 chip is not yet supported by Firebase Unity SDK.

Adding support is on the internal roadmap and you can track the status in the following issue:
https://github.com/firebase/quickstart-unity/issues/1100

There are also some comments in that issue of other firebase users building from source on the M1 and sharing generated bundles, so you could try to see if those work as a temporary workaround.

**Comment by sableangle** [comment]
I don't think it is similar issue to #1100.
Due to I can use Firebase SDK and everything works great, when I was using Unity 2020.3.x and the other thing on the environment is same as current one.
(Same machine, same OS, same Unity project, same python install status, different Unity Editor)

But once I upgrade project to Unity 2021.2, the issue comes.
(And after delete the Firebase.Editor.dll, the issue gones)

Since during the whole testing the only thing change is Unity Editor version, so I would say the Firebase SDK may has something wrong under Unity 2021.2.

**Comment by sableangle** [comment]
Hello, I am researching on this issue these days.
And have some new discovery, please see #1232, I'm not sure the issue is the same issue or not, so I create a new GitHub issue.

I don't have Intel Mac right now, I guess if Intel Mac can reproduce the problem on macOS 12.3, then the issue should positioned as the removing of Python2 on new OS version and the usage of Python2 inside the Firebase Editor C# script.

**Comment by ZhouKes** [comment]
solved :

https://github.com/techyworm10/firebase-unity-sdk-editor-python-fix

**Comment by chkuang-g** [comment]
I'm afraid that this might not be fixed by fixing `Firebase.Editor.dll` and seems to only occurs with Unity 2021.2.

Here is the way I reproduced this.
1. Use Unity 2021.2.17f1
2. Create an empty project
3. Install EDM4U and FirebaseApp (either .tgz or .unitypackage)
4. Patch the `Firebase.Editor.dll` with python fix (we have an internal build that fixes that issue)
5. Build an iOS build, or Android build.

Seems like the following log will always appears, even after clearing the log.
```
Internal build system error. Backend exited with code 2.
STDOUT:
Finished compiling graph: 245 nodes, 2781 flattened edges (2781 ToBuild, 0 ToUse), maximum node priority 99
STDERR:
tundra: error: Unexpected stdin
```

After initiating the build, Unity editor logs the following:
```
Build completed with a result of 'Failed' in 1 seconds (631 ms)
UnityEngine.StackTraceUtility:ExtractStackTrace () (at /Users/bokken/buildslave/unity/build/Runtime/Export/Scripting/StackTrace.cs:37)
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:LogError (object)
UnityEditor.BuildPlayerWindow/DefaultBuildMethods:BuildPlayer (UnityEditor.BuildPlayerOptions) (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindowBuildMethods.cs:192)
UnityEditor.BuildPlayerWindow:CallBuildMethods (bool,UnityEditor.BuildOptions) (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindowBuildMethods.cs:94)
UnityEditor.BuildPlayerWindow:GUIBuildButtons (UnityEditor.Modules.IBuildWindowExtension,bool,bool,bool,UnityEditor.Build.BuildPlatform,UnityEditor.Modules.IBuildPostprocessor) (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindow.cs:1171)
UnityEditor.BuildPlayerWindow:ShowBuildTargetSettings () (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindow.cs:1017)
UnityEditor.BuildPlayerWindow:OnGUI () (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindow.cs:517)
UnityEditor.HostView:InvokeOnGUI (UnityEngine.Rect) (at /Users/bokken/buildslave/unity/build/Editor/Mono/HostView.cs:448)
UnityEditor.DockArea:DrawView (UnityEngine.Rect) (at /Users/bokken/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:387)
UnityEditor.DockArea:OldOnGUI () (at /Users/bokken/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:378)
UnityEngine.UIElements.IMGUIContainer:DoOnGUI (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,bool,UnityEngine.Rect,System.Action,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:351)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,System.Action,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:693)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,System.Action,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:665)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:658)
UnityEngine.UIElements.IMGUIContainer:SendEventToIMGUIRaw (UnityEngine.UIElements.EventBase,bool,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:618)
UnityEngine.UIElements.IMGUIContainer:SendEventToIMGUI (UnityEngine.UIElements.EventBase,bool,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:607)
UnityEngine.UIElements.IMGUIContainer:HandleEvent (UnityEngine.UIElements.EventBase) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:535)
UnityEngine.UIElements.CallbackEventHandler:HandleEventAtTargetPhase (UnityEngine.UIElements.EventBase) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/Events/EventHandler.cs:151)
UnityEngine.UIElements.MouseCaptureDispatchingStrategy:DispatchEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/Events/MouseCaptureDispatchingStrategy.cs:88)
UnityEngine.UIElements.EventDispatcher:ApplyDispatchingStrategies (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:381)
UnityEngine.UIElements.EventDispatcher:ProcessEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:344)
UnityEngine.UIElements.EventDispatcher:ProcessEventQueue () (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:306)
UnityEngine.UIElements.EventDispatcher:OpenGate () (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:270)
UnityEngine.UIElements.EventDispatcherGate:Dispose () (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:75)
UnityEngine.UIElements.EventDispatcher:ProcessEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:372)
UnityEngine.UIElements.EventDispatcher:Dispatch (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel,UnityEngine.UIElements.DispatchMode) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:222)
UnityEngine.UIElements.BaseVisualElementPanel:SendEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.DispatchMode) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/Panel.cs:419)
UnityEngine.UIElements.UIElementsUtility:DoDispatch (UnityEngine.UIElements.BaseVisualElementPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:468)
UnityEngine.UIElements.UIElementsUtility:UnityEngine.UIElements.IUIElementsUtility.ProcessEvent (int,intptr,bool&) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:211)
UnityEngine.UIElements.UIEventRegistration:ProcessEvent (int,intptr) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:74)
UnityEngine.UIElements.UIEventRegistration/<>c:<.cctor>b__1_2 (int,intptr) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:28)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /Users/bokken/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:189)

(Filename: /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindowBuildMethods.cs Line: 192)

UnityEditor.BuildPlayerWindow+BuildMethodException: Error building Player because scripts have compile errors in the editor
  at UnityEditor.BuildPlayerWindow+DefaultBuildMethods.BuildPlayer (UnityEditor.BuildPlayerOptions options) [0x002cc] in /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindowBuildMethods.cs:193 
  at UnityEditor.BuildPlayerWindow.CallBuildMethods (System.Boolean askForBuildLocation, UnityEditor.BuildOptions defaultBuildOptions) [0x00080] in /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindowBuildMethods.cs:94 
UnityEngine.StackTraceUtility:ExtractStackTrace () (at /Users/bokken/buildslave/unity/build/Runtime/Export/Scripting/StackTrace.cs:37)
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:LogError (object)
UnityEditor.BuildPlayerWindow:CallBuildMethods (bool,UnityEditor.BuildOptions) (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindowBuildMethods.cs:99)
UnityEditor.BuildPlayerWindow:GUIBuildButtons (UnityEditor.Modules.IBuildWindowExtension,bool,bool,bool,UnityEditor.Build.BuildPlatform,UnityEditor.Modules.IBuildPostprocessor) (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindow.cs:1171)
UnityEditor.BuildPlayerWindow:ShowBuildTargetSettings () (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindow.cs:1017)
UnityEditor.BuildPlayerWindow:OnGUI () (at /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindow.cs:517)
UnityEditor.HostView:InvokeOnGUI (UnityEngine.Rect) (at /Users/bokken/buildslave/unity/build/Editor/Mono/HostView.cs:448)
UnityEditor.DockArea:DrawView (UnityEngine.Rect) (at /Users/bokken/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:387)
UnityEditor.DockArea:OldOnGUI () (at /Users/bokken/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:378)
UnityEngine.UIElements.IMGUIContainer:DoOnGUI (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,bool,UnityEngine.Rect,System.Action,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:351)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,UnityEngine.Matrix4x4,UnityEngine.Rect,System.Action,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:693)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,System.Action,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:665)
UnityEngine.UIElements.IMGUIContainer:HandleIMGUIEvent (UnityEngine.Event,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:658)
UnityEngine.UIElements.IMGUIContainer:SendEventToIMGUIRaw (UnityEngine.UIElements.EventBase,bool,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:618)
UnityEngine.UIElements.IMGUIContainer:SendEventToIMGUI (UnityEngine.UIElements.EventBase,bool,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:607)
UnityEngine.UIElements.IMGUIContainer:HandleEvent (UnityEngine.UIElements.EventBase) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:535)
UnityEngine.UIElements.CallbackEventHandler:HandleEventAtTargetPhase (UnityEngine.UIElements.EventBase) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/Events/EventHandler.cs:151)
UnityEngine.UIElements.MouseCaptureDispatchingStrategy:DispatchEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/Events/MouseCaptureDispatchingStrategy.cs:88)
UnityEngine.UIElements.EventDispatcher:ApplyDispatchingStrategies (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel,bool) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:381)
UnityEngine.UIElements.EventDispatcher:ProcessEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:344)
UnityEngine.UIElements.EventDispatcher:ProcessEventQueue () (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:306)
UnityEngine.UIElements.EventDispatcher:OpenGate () (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:270)
UnityEngine.UIElements.EventDispatcherGate:Dispose () (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:75)
UnityEngine.UIElements.EventDispatcher:ProcessEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:372)
UnityEngine.UIElements.EventDispatcher:Dispatch (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.IPanel,UnityEngine.UIElements.DispatchMode) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/EventDispatcher.cs:222)
UnityEngine.UIElements.BaseVisualElementPanel:SendEvent (UnityEngine.UIElements.EventBase,UnityEngine.UIElements.DispatchMode) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/Panel.cs:419)
UnityEngine.UIElements.UIElementsUtility:DoDispatch (UnityEngine.UIElements.BaseVisualElementPanel) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:468)
UnityEngine.UIElements.UIElementsUtility:UnityEngine.UIElements.IUIElementsUtility.ProcessEvent (int,intptr,bool&) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:211)
UnityEngine.UIElements.UIEventRegistration:ProcessEvent (int,intptr) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:74)
UnityEngine.UIElements.UIEventRegistration/<>c:<.cctor>b__1_2 (int,intptr) (at /Users/bokken/buildslave/unity/build/ModuleOverrides/com.unity.ui/Core/UIElementsUtility.cs:28)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&) (at /Users/bokken/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:189)

(Filename: /Users/bokken/buildslave/unity/build/Editor/Mono/BuildPlayerWindowBuildMethods.cs Line: 99)
```

There are some online posts but nothing really works for my case.
https://forum.unity.com/threads/how-do-i-resolve-this.1153742/

Will need to ask the team to look into this.

**Comment by chkuang-g** [comment]
Seems like this issue is not unique to Firebase SDK. 

I tried Facebook Unity SDK in an empty project and got the same error.

**Comment by saxahan** [comment]
I've got this build error also.
Upgraded to v8.9.0 today, expected python macos problem resolved; however, nothing changed :(

**Comment by sableangle** [comment]
With EDM4U 1.2.169 + Python patched Firebase 8.8.0 this issue will not happen.
But EDM4U 1.2.170 + official Firebase 8.9.0 the issue back.

All tested on Unity 2021.2.X

So Unity has issue, and some part of the EDM4U + Firebase will trigger this issue.
And the Python issue patched Firebase 8.8.0 are just no problem?

That's all clues we have.

**Comment by chkuang-g** [comment]
@sableangle 

Odd... most of the change in EDM4U `1.2.170` are related to iOS Resolver.
https://github.com/googlesamples/unity-jar-resolver/releases/tag/v1.2.170

What happens if you downgrade EDM4U to `1.2.169`
You will need to remove `Assets/ExternalDependencyManager` first before installing the older version, if you imported it through `.unitypackage`.

**Comment by jonsimantov** [comment]
> With EDM4U 1.2.169 + Python patched Firebase 8.8.0 this issue will not happen. But EDM4U 1.2.170 + official Firebase 8.9.0 the issue back.

When you attempt using the 8.9.0 SDK, does it print anything in your error log about finding a python executable?

(The updated version should log an error if it was unable to find Python.)

**Comment by sableangle** [comment]
> When you attempt using the 8.9.0 SDK, does it print anything in your error log about finding a python executable?
> 
> (The updated version should log an error if it was unable to find Python.)

No python relative error logs found when using 8.9.0 SDK in my case.

**Comment by sableangle** [comment]
Hello guys!

I have some new infos.

Following this:
> With EDM4U 1.2.169 + Python patched Firebase 8.8.0 this issue will not happen. But EDM4U 1.2.170 + official Firebase 8.9.0 the issue back.
> 
> All tested on Unity 2021.2.X
> 
> So Unity 2021 may has issue, and some part of the EDM4U + Firebase will trigger this issue. And the Python issue patched Firebase 8.8.0 are just no problem?
> 
> That's all clues we have.

My current project is EDM4U 1.2.170 + official Firebase 8.9.0.
The issue back and also error message back
> Internal build system error. Backend exited with code 2.

But after I delete the Firebase.Editor.dll(delivery by official sdk 8.9.0), the issue just disappeared!

And then I also tried replace the Firebase.Editor.dll file by the one I was using in SDK 8.8.0(3rd party python patched), the issue doesn't happen too.
This one I used:
(https://github.com/techyworm10/firebase-unity-sdk-editor-python-fix) 

----

So, here is the test result we have: 


Combination | Status
-- | --
Empty Project | ✅
EDM4U alone | ✅
EDM4U + FirebaseAppDependencies.xml | ✅
Firebase + EDM4U | ❌
Facebook + EDM4U | ❌

----above part  just copy/paste from chkuang-g

Id | Combination | Status
-- | -- | --
1 | Firebase Official 8.9.0 + EDM4U 1.2.169 | ❌
2 | Firebase Official 8.9.0 + EDM4U 1.2.170 | ❌
3 | Firebase 8.8.0 (with techyworm10's python patch Firebase.Editor.dll ) + EDM4U 1.2.169 | ✅
4 | Firebase 8.9.0 (with techyworm10's python patch Firebase.Editor.dll ) + EDM4U 1.2.170 | ✅
5 | Firebase 8.9.0 (Delete Firebase.Editor.dll ) + EDM4U 1.2.170 | ✅
6 | Firebase 8.9.0 (Delete Firebase.Editor.dll ) + EDM4U 1.2.169 | ✅
7 | Firebase 8.9.0 (Delete Firebase.Editor.dll ) + EDM4U 1.2.169 | ✅

And only the Id 1 prints error log about cannot find a python executable

**Comment by AndreySkyFoxSidorov** [comment]
In IOS "Internal build system error. Backend exited with code" error after last update of the Operating system and import Firebase SDK 8.9.0.

Unity 2021.2.5 - 2021.2.15, on an older 2021.1.Х version everything is ok.
External Dependency Manager 1.2.168, 1.2.169, 1.2.170.
Firebase SDK 8.9.0
macOS 12.3.1 (21E258)
MacBook Pro(MacBookPro12,1 Dual-Core Intel Core i5)
Are there any ideas how to fix this?

**Comment by Fangh** [comment]
Uniy 2021.3 + Firebase 8.9.0 + Apple M1

```
DllNotFoundException: FirebaseCppApp-8_9_0 assembly:<unknown assembly> type:<unknown type> member:(null)
Firebase.AppUtilPINVOKE+SWIGExceptionHelper..cctor () (at Z:/tmp/tmp.GOobKx5vLv/firebase/app/client/unity/proxy/AppUtilPINVOKE.cs:117)
Rethrow as TypeInitializationException: The type initializer for 'SWIGExceptionHelper' threw an exception.
Firebase.AppUtilPINVOKE..cctor () (at Z:/tmp/tmp.GOobKx5vLv/firebase/app/client/unity/proxy/AppUtilPINVOKE.cs:137)
Rethrow as TypeInitializationException: The type initializer for 'Firebase.AppUtilPINVOKE' threw an exception.
Firebase.AppUtil.SetLogFunction (Firebase.LogUtil+LogMessageDelegate arg0) (at Z:/tmp/tmp.GOobKx5vLv/firebase/app/client/unity/proxy/AppUtil.cs:64)
Firebase.LogUtil..ctor () (at Z:/tmp/tmp.GOobKx5vLv/firebase/app/client/unity/proxy/LogUtil.cs:81)
Firebase.LogUtil..cctor () (at Z:/tmp/tmp.GOobKx5vLv/firebase/app/client/unity/proxy/LogUtil.cs:37)
Rethrow as TypeInitializationException: The type initializer for 'Firebase.LogUtil' threw an exception.
Firebase.FirebaseApp..cctor () (at Z:/tmp/tmp.GOobKx5vLv/firebase/app/client/unity/proxy/FirebaseApp.cs:67)
Rethrow as TypeInitializationException: The type initializer for 'Firebase.FirebaseApp' threw an exception.
FirebaseChecker.Start () (at Assets/Scripts/FirebaseChecker.cs:13)
```

**Comment by Fangh** [comment]
Unity 2021.3 (Intel version) + Firebase 8.9.0 + Apple M1 + Android Build = working perfectly

**Comment by alvincrisuy** [comment]
<img width="328" alt="Screen Shot 2022-04-22 at 3 43 08 PM" src="https://user-images.githubusercontent.com/7260700/164635983-d4a51047-c622-4043-ae09-185ba62d1e34.png">
I also have encountered this issue and nothing still works as expected.

**Comment by lukaslabaj** [comment]
I have same issue `Internal build system error. Backend exited with code 2.` Unity 2020.2.10, any solution?

**Comment by korober** [comment]
Unity 2021.3 (Intel version) + Firebase 8.9.0  same issue (backend exited with code 2)
Fixed [with the workaround.](https://github.com/firebase/firebase-unity-sdk/issues/154#issuecomment-1068879556)

**Comment by AlmostMatt** [comment]
Based on my testing so far, this issue does not occur when using the just-released version 8.10.0 of the Firebase Unity SDK
I was able to just add the relevant 8.10.0 packages to a project that had been failing, restarted Unity, and then builds succeeded.

I'll continue to look into why the issue occurs in 8.9.0, but in the meantime using 8.10.0 is a possible workaround

**Comment by AlmostMatt** [comment]
The relevant change that fixes this in 8.10.0 seems to be the change to try Python3 before trying Python (https://github.com/firebase/firebase-unity-sdk/pull/249). I tried to debug further, but it's still not entirely clear to me why trying Python is causing it to fail despite the exception handling present in the code.

But since it seems to be working in 8.10.0 for now, I am going to close this issue and recommend updating to 8.10.x if this issue occurs.

If anyone encounters this issue despite using firebase unity SDK 8.10.x or later, please reopen.
